### PR TITLE
Fix parsing of nested verbatim interpolated strings started with @$

### PR DESCRIPTION
**Pull request by rstarkov**
When parsing an interpolated string, the function that scans to find the end of an interpolated hole, `ScanInterpolatedStringLiteralHoleBalancedText`, does not correctly account for the possibility of a nested string starting with @$.

This PR fixes this by copying the code already present for the $@ case just a few `case`s earlier.

I've included a unit test which fails without this patch.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/roslyn?pullRequest=44924) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/roslyn?pullRequest=44924) before we can accept your contribution.<br/><br/>:x: rstarkov [sign now](https://cla.dotnetfoundation.org/dotnet/roslyn?pullRequest=44924)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/roslyn?pullRequest=44924) it.</sub>

File: src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs [hunk 1]
```diff
@@ -600,6 +600,26 @@ private void ScanInterpolatedStringLiteralHoleBalancedText(char endingChar, bool
                                 ScanInterpolatedStringLiteralNestedVerbatimString();
                                 continue;
                             }
+                            else if (lexer.TextWindow.PeekChar(1) == '$' && lexer.TextWindow.PeekChar(2) == '"')
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs:603]: Thanks for finding and fixing this.

We probably need to add a LangVersion check/diagnostic though (`                       CheckFeatureAvailability(MessageID.IDS_FeatureAltInterpolatedVerbatimStrings)`)

**Comment by rstarkov** [comment]
Added the language version diagnostic and unit test for it. This test style seems superior so I've deleted the old test.

Note that the error location is the `$` of the `@$` sequence. Feels like ideally it should be the `@` but I couldn't find an easy way to do that. Let me know if you want me to address this?

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for a second review of community PR. Thanks

**Comment by jcouv** [comment]
Merged/squashed. Thanks @rstarkov for the contribution! :-)

**Pull request merged**